### DISCUSSION TITLE
Toggle in 'now playing' menu that preserves volume on window inactive

### DISCUSSION THREAD

**Discussion by Konomi-io** [ppy/osu#30182] [Ideas] [answered]
When normally playing the game, having the game reduce volume when tabbed out is really nice, but sometimes you want to use osu as a music player and keep the volume the same whether its active or not. 

The suggestion to add a toggle in settings has been denied multiple times (#15143 and #21852), as it's possible to do it by settings the "window inactive" audio setting to 100%. This is sufficient for those who change infrequently, but otherwise can be clunky as you have to search for the setting every time, and remember the value you had it set to when you want to return it.

Having a quick toggle button in the top right 'now playing' menu would make things easy, as this is the most relevant place to do so. An extra setting entry is not needed as it can just affect the slider, then automatically be disabled if the slider is adjusted. I believe this is distinct enough from the other suggestions and useful enough to justify a post

**Comment by bdach** [comment, marked as answer]
> you want to use osu as a music player

It is not one full stop.

**Reply by Konomi-io** [reply]
i feel that this response is a bit closed minded. i dont see a problem with accommodating common use cases for the platform, especially when a small qol feature like this would impact the scope so little.

osu can still implement some helpful features without "becoming a music player", and i think some small things like this are more than warranted enough. shuffle was recently added to the now playing menu, and it can be easily argued that that is a "music player" feature as well, so where do we even draw the line?

i understand that there are a lot of things going on with osu, and copyright is not something to messed with. i was never expecting this idea to be implemented soon, if at all, but such a hasty generalisation seems damning

**Reply by Konomi-io** [reply]
btw my intent is not to be argumentative or arrogant, so sorry if i come off that way. i'm assuming this wont go anywhere from here so i'll drop it, i just hope there's a possibility for my point to be considered in the general case

**Reply by bdach** [reply]
> shuffle was recently added to the now playing menu, and it can be easily argued that that is a "music player" feature as well, so where do we even draw the line?

Shuffle was only added to fix a longstanding bug wherein music playback order wasn't randomised, which is a departure from stable.

I don't want to see fifteen slider bars and thirty toggles in the volume section. Four separate volume controls should be enough.
